src/main.rs

```diff
@@ -1,25 +1,34 @@
 #![feature(proc_macro_hygiene, decl_macro)]
 
-use rocket::{self, routes, Config};
 use std::env;
+use rocket::{self, routes, Config};
 
-mod controllers;
 mod domain;
+mod items_controller;
+mod pudeuko_service;
 mod dropbox_client;
-mod service;
 
 fn create_configuration() -> Config {
-    let mut config = Config::active().expect("could not load configuration");
+    let mut config = Config::active().expect("Could not load configuration");
     if let Ok(port_str) = env::var("PORT") {
-        let port = port_str.parse().expect("could not parse PORT");
+        let port = port_str.parse().expect("Could not parse PORT");
         config.set_port(port);
     }
     config
 }
 
 fn main() {
+    let dropbox_token = env::var("DROPBOX_TOKEN").expect("You have to provide a Dropbox API token");
+    let dropbox_client = dropbox_client::DropboxClient::new(dropbox_token);
+    let pudeuko_service = pudeuko_service::PudeukoService::new(dropbox_client);
+
     let config = create_configuration();
     rocket::custom(config)
-        .mount("/items", routes![controllers::get_items, controllers::post_item, controllers::get_item ])
+        .manage(pudeuko_service)
+        .mount("/items", routes![
```

Comment:
Patrząc na tę aplikację widzę w zasadzie takie moduły:
- config (czyta konfigurację - tutaj mogłoby być rozróżnianie środowisk dev/test/prod)
- pudeuko (core aplikacji, udostępnia funkcjonalności, z których korzystają "frontendy" czyli moduł api lub ewentualny moduł cli, gdybyś chciał zrobić apkę command line'ową)
- api (udostępnia funkcjonalności modułu storage, przez api)

Patrząc natomiast na kod nie widzę takiego jasnego podziału. To chyba najlepiej widać w `domain.rs`, w którym masz wrzucone modele modułów storage i api. Proponowałbym taki układ pliczków/odpowiedzialnośći:

```
main.rs - importuje moduły config, storage, api i tworzy z nich aplikację (powinno wyjść tylko kilka linijek kodu)
config.rs - kod, ktory pozwala wczytac configi
pudeuko/
- mod.rs - jest fasadą/interfejsem całego modułu storage, reeksportuje funkcje konkretnej implementacji (w tym wypadku PudeukoService, może reeksportować też potrzebne modele z domain.rs/model.rs) Pewnie ostatecznie będzie tam funkcja, która przyjmuje obiekt config i zwraca PudeukoService
- dropbox_client.rs  - niedostępny z zewnątrz modułu
- pudeuko_service.rs - pewnie wymagalby nieco dopracowania
- domain.rs / model.rs - zawiera modele używane w tym module oraz zwracane przez moduł (np. Item), żadnego DTO
api/
- mod.rs - funkcja, która przyjmuje przyjmuje pudeuko service stworzony i buduje i uruchamia serwer rocketowy
- handlers.rs - to co teraz jest w items_controller.rs
- domain.rs / model.rs - modele API, tudzież DTOsy
```

Wtedy patrząc na listę plików i katalogów już lepiej widać jakie "grube" części ma aplikacja. Jeśli tego nie czujesz to daj znać - mogę spróbować zrobić jakiegoś PRa z takim refactorem i zobaczysz o co mi chodzi.

Uwaga: masz mapowanie pomiędzy DTOsem a Itemem, ten kod powinien być też tam gdzie model ContentDTO. Mój zamysł jest taki, że storage nie musi nic wiedzieć o API (a więc tym bardziej o DTOsie) i gdybyś chciał zrobić apkę pudełko-cli do obsługi z linii komend, to wystarczy, że dorzucisz moduł cli, który skorzysta z modułu storage, a moduł api zwyczajnie wywalić.